cloudinit/cmd/devel/logs.py

```diff
@@ -196,131 +171,207 @@ def _stream_command_output_to_file(
+            copy(path, out_dir)
+            LOG.debug("collected file: %s", path)
+        else:
+            LOG.trace("sensitive file %s was not collected", path)
     else:
         LOG.trace("file %s did not exist", path)
 
 
-def _collect_installer_logs(log_dir: str, include_userdata: bool) -> None:
+def _collect_installer_logs(
+    log_dir: pathlib.Path, include_sensitive: bool
+) -> None:
     """Obtain subiquity logs and config files."""
     for src_file in INSTALLER_APPORT_FILES:
-        destination_dir = pathlib.Path(log_dir + src_file.path).parent
-        if not destination_dir.exists():
-            ensure_dir(str(destination_dir))
-        _collect_file(src_file.path, str(destination_dir))
-    if include_userdata:
+        destination_dir = pathlib.Path(log_dir, src_file.path[1:]).parent
+        _collect_file(
+            pathlib.Path(src_file.path),
+            destination_dir,
+            include_sensitive=True,  # Because this function does check
+        )
+    if include_sensitive:
         for src_file in INSTALLER_APPORT_SENSITIVE_FILES:
-            destination_dir = pathlib.Path(log_dir + src_file.path).parent
-            if not destination_dir.exists():
-                ensure_dir(str(destination_dir))
-            _collect_file(src_file.path, str(destination_dir))
+            destination_dir = pathlib.Path(log_dir, src_file.path[1:]).parent
+            _collect_file(
+                pathlib.Path(src_file.path),
+                destination_dir,
+                include_sensitive=True,  # Because this function does check
+            )
 
 
-def _collect_version_info(log_dir: str) -> None:
+def _collect_version_info(log_dir: pathlib.Path) -> None:
+    """Include cloud-init version and dpkg version in the logs."""
     version = _write_command_output_to_file(
         cmd=["cloud-init", "--version"],
-        file_path=pathlib.Path(log_dir, "version"),
+        file_path=log_dir / "version",
         msg="cloud-init --version",
     )
     dpkg_ver = _write_command_output_to_file(
         cmd=["dpkg-query", "--show", "-f=${Version}\n", "cloud-init"],
-        file_path=pathlib.Path(log_dir, "dpkg-version"),
+        file_path=log_dir / "dpkg-version",
         msg="dpkg version",
     )
     if not version:
         version = dpkg_ver or "not-available"
 
 
-def _collect_system_logs(log_dir: str) -> None:
-    _stream_command_output_to_file(
-        cmd=["dmesg"],
-        file_path=pathlib.Path(log_dir, "dmesg.txt"),
-        msg="dmesg output",
-    )
+def _collect_system_logs(
+    log_dir: pathlib.Path, include_sensitive: bool
+) -> None:
+    """Include dmesg and journalctl output in the logs."""
+    if include_sensitive:
+        _stream_command_output_to_file(
+            cmd=["dmesg"],
+            file_path=log_dir / "dmesg.txt",
+            msg="dmesg output",
+        )
     _stream_command_output_to_file(
         cmd=["journalctl", "--boot=0", "-o", "short-precise"],
```

Comment:
If the instance has rebooted (which is configurable in multiple parts of cloud-config), or in the case of debugging subiquity, the journal from the previous boot could be particularly helpful - do you think that we want to include the `journalctl -b -1` as `journal-last.txt`?